frontends/bmd/src/pages/poll_worker_screen.tsx

```diff
@@ -116,39 +136,35 @@ function PrecinctScannerTallyReportModal({
   precinctScannerTally,
   pollworkerAuth,
   machineConfig,
-  isPollsOpen,
-  togglePollsOpen,
+  pollsState,
+  updatePollsState,
   onClose,
+  logger,
 }: {
   electionDefinition: ElectionDefinition;
   precinctScannerTally: PrecinctScannerCardTally;
   pollworkerAuth: InsertedSmartcardAuth.PollWorkerLoggedIn;
   machineConfig: MachineConfig;
-  isPollsOpen: boolean;
-  togglePollsOpen: VoidFunction;
+  pollsState: PollsState;
+  updatePollsState: (pollsState: PollsState) => void;
   onClose: VoidFunction;
+  logger: Logger;
 }) {
   const [modalState, setModalState] =
     useState<PrecinctScannerTallyReportModalState>('initial');
 
-  const parties = getPartyIdsInBallotStyles(electionDefinition.election);
   const precinctScannerTallyInformation =
     precinctScannerTally &&
     parsePrecinctScannerTally(
       precinctScannerTally,
-      electionDefinition.election,
-      parties
+      electionDefinition.election
     );
-
-  function togglePollsToMatchReport() {
-    if (
-      precinctScannerTally &&
-      precinctScannerTally.isPollsOpen !== isPollsOpen
-    ) {
-      togglePollsOpen();
-    }
-  }
-
+  const precinctScannerPollsState = getPollsTransitionDestinationState(
+    precinctScannerTally.pollsTransition
+  );
+  const [willUpdatePollsToMatchScanner] = useState(
```

Comment:
why is this a state variable if it can't change?

Reply:
I'm making it a state variable so it _does not_ change. After the polls update, the `pollsState` will be different and then `willUpdatePollsToMatchScanner` will be different. This way it is the same for every render until the component is unmounted. Is there a more appropriate way to do this?